packages/idyll-compiler/src/lexer.js

```diff
@@ -109,81 +147,132 @@ const lex = function(options) {
     return ['INLINE_CODE'].concat(formatToken(text.trim()));
   });
 
-  lexer.addRule(/[\s\n]*(#{1,6})\s*([^\n\[]+)[\n\s]*/gm, function(lexeme, hashes, text) {
+  lexer.addRule(/[\s\n]*(#{1,6})\s*([^\n\[]+)[\n\s]*/gm, function(
+    lexeme,
+    hashes,
+    text
+  ) {
     if (this.reject) return;
     updatePosition(lexeme);
-    return ['BREAK', 'HEADER_' + hashes.length].concat(recurse(text, { skipLists: true })).concat(['HEADER_END']);
+    return ['BREAK', 'HEADER_' + hashes.length]
+      .concat(recurse(text, { skipLists: true }))
+      .concat(['HEADER_END']);
   });
 
   lexer.addRule(/[\s\n]*>\s*([^\n\[]+)[\n\s]*/gm, function(lexeme, text) {
     if (this.reject) return;
     updatePosition(lexeme);
-    return ['BREAK', 'QUOTE_START'].concat(recurse(text, { skipLists: true })).concat(['QUOTE_END']);
+    return ['BREAK', 'QUOTE_START']
+      .concat(recurse(text, { skipLists: true }))
+      .concat(['QUOTE_END']);
   });
 
-
-  lexer.addRule(/\*\*([^\s\n][^\*]*[^\s\n])\*\*(\s*)/g, function(lexeme, text, trailingSpace) {
+  lexer.addRule(/\*\*([^\s\n][^\*]*[^\s\n])\*\*(\s*)/g, function(
+    lexeme,
+    text,
+    trailingSpace
+  ) {
     this.reject = inComponent;
     if (this.reject) return;
     updatePosition(lexeme);
-    var ret = ['STRONG'].concat(recurse(text, { skipLists: true })).concat(['STRONG_END']);
+    var ret = ['STRONG']
+      .concat(recurse(text, { skipLists: true }))
+      .concat(['STRONG_END']);
     if (trailingSpace) {
-      ret = ret.concat(['WORDS']).concat(formatToken(trailingSpace))
+      if (shouldBreak(trailingSpace)) {
+        ret = ret.concat(['BREAK']);
+      } else {
+        ret = ret.concat(['WORDS']).concat(formatToken(trailingSpace));
+      }
```

Comment:
The PR for this is a bit messy due to this being the first time running prettier on the lexer code, but the relevant changes are here, and are mirrored in the other rules for `strong` and `em`.